pennylane/noise/conditionals.py

```diff
@@ -471,46 +624,125 @@ def partial_wires(operation, *args, **kwargs):
+
+    >>> func = qml.noise.partial_wires(qml.expval(qml.Z(0)))
+    >>> func(qml.RX(1.2, wires=[9]))
+    qml.expval(qml.Z(9))
     """
+    is_meas_class = isinstance(operation, qml.measurements.MeasurementProcess)
+    is_meta_class = isinstance(operation, (qml.ops.Adjoint, qml.ops.Controlled))
+    is_mappable, map_op = False, None
+
     if not callable(operation):
         if args:
             raise ValueError(
                 "Args cannot be provided when operation is an instance, "
                 f"got operation = {operation} and args = {args}."
             )
+        if not kwargs:
+            is_mappable, map_op = True, operation
         args, metadata = getattr(operation, "_flatten")()
+        is_metadata_flat = is_meas_class or isinstance(operation, (qml.ops.Controlled))
         if len(metadata) > 1:
-            kwargs = {**dict(metadata[1]), **kwargs}
+            kwargs = {**dict(metadata[1] if not is_metadata_flat else metadata), **kwargs}
         operation = type(operation)
+    elif operation in _MEAS_FUNC_MAP:
+        is_meas_class = True
+        operation = _MEAS_FUNC_MAP[operation]
+    elif operation in (
+        qml.adjoint,
+        qml.ctrl,
+    ):
+        is_meta_class = True
+        operation = {qml.adjoint: qml.ops.Adjoint, qml.ctrl: qml.ops.Controlled}[operation]
 
     fsignature = signature(getattr(operation, "__init__", operation)).parameters
     parameters = list(fsignature)[int("self" in fsignature) :]
     arg_params = {**dict(zip(parameters, args)), **kwargs}
 
-    if "wires" in arg_params:  # Ensure we don't include wires arg
-        arg_params.pop("wires")
+    arg_wires = arg_params.pop("wires", None)  # Ensure we don't include wires in name
+    if is_meas_class and "op" in arg_params:
+        arg_params["obs"] = arg_params.pop("op")
+
+    if is_meta_class and "op" in arg_params:
+        arg_params["base"] = arg_params.pop("op")
 
-    op = partial(operation, **{**arg_params, **kwargs})
+    if operation == qml.ops.Controlled and "control" in arg_params:
+        arg_params["control_wires"] = arg_params.pop("control")
 
     op_name = f"{operation.__name__}("
-    for key, val in op.keywords.items():
-        op_name += f"{key}={val}, "
-    op_name = op_name[: -2 if len(op.keywords) else -1] + ")"
+    for key, val in arg_params.copy().items():
+        if key in parameters:
+            op_name += f"{key}={val}, "
+        else:  # pragma: no cover
+            del arg_params[key]
+    op_name = op_name[:-2] + ")" if len(arg_params) else op_name[:-1]
 
     @_rename(op_name)
-    def _partial_op(wires, **model_kwargs):  # pylint: disable = unused-argument
+    def _partial(wires=None, **partial_kwargs):
         """Wrapper function for partial_wires"""
-        wires = getattr(wires, "wires", None) or (
-            [wires] if isinstance(wires, (int, str)) else list(wires)
-        )
-        return op(wires=wires)
+        op_args = arg_params
```

Comment:
The following addition seems too tedious, is there a consistency change we can make elsewhere, e.g., replacing "H" with "obs" in ShadowExpvalMP, that helps us here?

Reply:
We can add it to the backlog, I've refactored the code to make this a little less tedious.